### DISCUSSION TITLE
NGINX and SocketIO Reverse Proxy 200 but "Unable to connect"

### DISCUSSION THREAD

**Discussion by pdtc** [miguelgrinberg/python-socketio#1332] [Q&A] [answered]
Hi Everyone,

I've been struggling with this for 2 days now so maybe someone can help here. This application logic works locally just fine.

I've written a python script that is essentially an LLM Chatbot that uses vector storage etc. it uses socketio to return the llm answer to a React frontend. The setup below produces 200 status codes on socket connection. However the 3rd call to socketio results in a **"4/python,"Unable to connect"** message, is anyone familiar with this?

Here's the server output with logging enabled:
````
4OlyCdNwpXhBfKZvAAAB: Sending packet OPEN data {'sid': '4OlyCdNwpXhBfKZvAAAB', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}

4OlyCdNwpXhBfKZvAAAB: Received packet MESSAGE data 0/python,

4OlyCdNwpXhBfKZvAAAB: Sending packet MESSAGE data 4/python,"Unable to connect"

4OlyCdNwpXhBfKZvAAAB: Received request to upgrade to websocket

4OlyCdNwpXhBfKZvAAAB: Received packet CLOSE data

4OlyCdNwpXhBfKZvAAAB: Failed websocket upgrade, no PING packet
````

The python server is running on gunicorn/eventlet: 
`gunicorn -b 127.0.0.1:6001 --worker-class eventlet -w 1 aibackend:app --reload`

The webserver is running nginx here's the config:

`
server {
    listen 80;
    server_name [redacted];

    root [redacted]

    location / {
        try_files $uri $uri/ =404;
    }

    location /socket.io {
        proxy_pass http://localhost:6001;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;

        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        proxy_set_header X-NginX-Proxy true;

        proxy_redirect off;
    }
}`

Here's the python code:
```
import queue
import threading

import eventlet
import socketio
from flask import Flask, jsonify
from flask_cors import CORS, cross_origin

sio = socketio.Server(
    cors_allowed_origins=[
        "http://localhost:3001",
        "[THIS IS THE ORIGIN URL]",
    ],
    logger=True,
    engineio_logger=True,
    sync_mode="eventlet",
)

app = Flask(__name__)
app.wsgi_app = socketio.WSGIApp(sio, app.wsgi_app)

socket_clients = {}

@sio.on("connect")
def handle_connect(sid, environ, auth):
    print("CONNECT", sid)
    client_id = sid
    socket_clients[client_id] = sid

@sio.on("message")
def handle_message(sid, data, data2):
    question = data
    chatHistory = data2
    client_id = sid

    threading.Thread(target=ctx_mgr).start()
    response = process_question(question, chatHistory)
    sio.emit(
        "response",
        {"message": convert_to_json_serializable(response)},
        room=client_id,
    )


@sio.on("disconnect")
def handle_disconnect(sid):
    print("DISCONNECT", sid)
    client_id = sid
    if client_id in socket_clients:
        del socket_clients[client_id]

// LLM CODE

if __name__ == "__main__":
    # Use eventlet to run Flask with SocketIO
    eventlet.wsgi.server(eventlet.listen(('', 8000)), app)`
```

**Comment by miguelgrinberg** [comment, marked as answer]
Your client is trying to connection to a namespace named `/python`:

    4OlyCdNwpXhBfKZvAAAB: Received packet MESSAGE data 0/python,
    4OlyCdNwpXhBfKZvAAAB: Sending packet MESSAGE data 4/python,"Unable to connect"

The server returns an error because it does not have any knowledge of this namespace.

**Reply by pdtc** [reply]
That worked! Thank you so much. I don't know why this works locally but needed the namespaces on live but I can't thank you enough!
